crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -1177,20 +1177,22 @@ impl PgManager {
                     Error::Internal("Error converting from Object to StakedSui".to_string())
                 })?;
 
-                // TODO this only does active / pending stake status, but not unstaked
+                // TODO this only does active / pending stake status, but not unstaked,
+                // unstaked does not really make sense here, fullnode tracks deleted objects
                 let status = if current_epoch_id >= stake_object.activation_epoch() {
-                    Some(StakeStatus::Active)
+                    StakeStatus::Active
                 } else {
-                    Some(StakeStatus::Pending)
+                    StakeStatus::Pending
                 };
                 let stake = Stake {
                     active_epoch_id: Some(stake_object.activation_epoch()),
                     estimated_reward: None, // TODO once we have a good working governance API, we should fix this
                     principal: Some(BigInt::from(stake_object.principal())),
-                    request_epoch_id: Some(stake_object.activation_epoch() - 1),
-                    status,
+                    request_epoch_id: Some(stake_object.activation_epoch().saturating_sub(1)),
```

Comment:
today, if activation_epoch is 0, then request_epoch_id = -2209834023784982738423 or something. I feel like setting the value to 1 would be confusing. Why not keep both at 0? And are we sure this isn't a breaking change?

Reply:
Not sure if this would be a breaking change or not, possibly yet. However, after talking with @bmwill , we pushed this fix in the governance API, because `request_epoch_id` is `u64`, so if activation_epoch is 0 it will panic with some overflow. The saturating sub will make sure it does not overflow, so basically if `activation_epoch = 0` => `request_epoch = 0`. The activation epoch to be 0 can happen at genesis.